arch/powerpc/kernel/eeh_sysfs.c

```diff
@@ -91,6 +91,23 @@ static ssize_t eeh_pe_state_store(struct device *dev,
 
 static DEVICE_ATTR_RW(eeh_pe_state);
 
+#ifdef CONFIG_PCI_IOV
+static ssize_t eeh_notify_resume_show(struct device *dev,
+				 struct device_attribute *attr, char *buf)
+{
+	struct pci_dev *pdev = to_pci_dev(dev);
+	struct eeh_dev *edev = pci_dev_to_eeh_dev(pdev);
+	int state;
+
+	if (!edev || !edev->pe)
+		return -ENODEV;
+
+	state = eeh_ops->notify_resume(pci_get_pdn(pdev));
+	return sprintf(buf, "0x%08x\n", state);
```

Comment:
%08x i believe is unsigned int? your state is int. Unsigned int cant hold negative values, so your notify_resume basically does a RTAS call right? RTAS calls can return negative values. 

http://elixir.free-electrons.com/linux/v4.14/source/arch/powerpc/include/asm/rtas.h#L21

Reply:
Now I am a little more confused about this one:


d or i | Signed decimal integer | 392
u | Unsigned decimal integer | 7235
o | Unsigned octal | 610
x | Unsigned hexadecimal integer | 7fa
X | Unsigned hexadecimal integer (uppercase) | 7FA

Does it matter to use a hex value when printing the data? tell me which one you want me to use?

Reply:
Just have both _store and _show. Steve will only call the _store and the return code passed back to him should be a file system return code. So use something like http://www-numi.fnal.gov/offline_software/srt_public_context/WebDocs/Errors/unix_system_errors.html as the return from _store if the notify_resume fails. You would need to store the state somewhere in your structures, then in the _show call do a return of a signed int non hex representation.

Reply:
ok. I need what the user wants and the mapping of those return codes. Because what you describe is me mapping back to error number in linux. 

Which to me does not allow the interface to grow. So tell me the mapping for these error 
codes from rtas:
-1: Hardware Error
-3: Parameter Error

translate to your errorno.

Reply:
eeh_notify_resume_show - should snprintf an unsigned int value of the rtas call return code.
eeh_notify_resume_store - should return linux system error codes. So like you did in your code if !edev or !edev->pe then return -ENODEV. I dont know what all the possibilites are of your rtas return code, so I cant translate for you. But obviously just pick stuff that makes sense or use 1 generic bad return code if the rtas call fails. If rtas rc = RTAS_BUSY use EBUSY... Etc. It doesn't really matter in the _store case because steve just cares whether or succeeds or not. He doesnt care what the actual return code is. So even if you just returned EPERM or w/e something generic it would be fine on failure of that resume call. On the _store on the other hand if for debug you wanted more specific value of the rtas return code then you snprintf the saved state from the _store call. This saved state should contain the actual RTAS value, and that should be a signed int since it can be negative.

Reply:
ok.

Reply:
Incorporated the changes. Let me know what you think.